1.19 Compatibillity fix for RefabricatedSlabs mod

CebbyS opened the pull request:
## Info
Compatibility fix for issue https://github.com/ModdingForBlockheads/KleeSlabs/issues/53 for RefabricatedSlabs mod

## Changes
SlabConverter
- Added a new method to the SlabConverter interface for compatibility fix

BlockBreakHandler
- Vector hit position calculated before asking for single slab state
- By doing so correct slab is used for block state property calculation
- Changed SlabConverter method calls to the new method

CebbyS committed:
SlabConverter
- Added a new method to the SlabConverter interface for compatibility fix

BlockBreakHandler
- Vector hit position calculated before asking for single slab state
- By doing so correct slab is used for block state property calculation
- Changed SlabConverter method calls to the new method

CebbyS committed:
SlabConverter
- Added a new method to the SlabConverter interface for compatibility fix

BlockBreakHandler
- Vector hit position calculated before asking for single slab state
- By doing so correct slab is used for block state property calculation
- Changed SlabConverter method calls to the new method

BlayTheNinth commented on shared/src/main/java/net/blay09/mods/kleeslabs/converter/SlabConverter.java:
```diff
@@ -1,14 +1,19 @@
 
     BlockState getSingleSlab(BlockState state, SlabType slabType);
 
+    default BlockState getSingleSlab(BreakBlockEvent event, SlabType slabType) {
```
I would rather not want to expose Balm's `BreakBlockEvent` as public API (even though SlabConverter itself isn't really public API right now). It would be better if the method just accepted all necessary parameters similar to how Vanilla does it in `Block` methods.

CebbyS replied:
I see, okay, let me try to do some changes. Thanks!

CebbyS committed:
Changed SlabConverter API.
Changed BreakBlockEvent to BlockState, Level, BlockPos, Player.
Updated subclasses that implements SlabConverter to the new API.

CebbyS committed:
Merge branch '1.19.x-refabslab-compatibility-fix' of https://github.com/CebbysMods/KleeSlabs into 1.18.x-refabslab-compatibility-fix

 Conflicts:
	shared/src/main/java/net/blay09/mods/kleeslabs/BlockBreakHandler.java
	shared/src/main/java/net/blay09/mods/kleeslabs/converter/SlabConverter.java

CebbyS commented:
Hi, I have implemented changes for both of the MR I have created.
